src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,64 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE_FOR_CANDIDATE = 35;
+    private static final int MIN_PERIODS_IN_UKRAINE = 10;
+    private static final String NATIONALITY = "Ukrainian";
+    private static final String DELIMITER = "-";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate.getAge() < MIN_AGE_FOR_CANDIDATE) {
```

Comment:
```suggestion
        return candidate.getAge() >= MIN_AGE_FOR_CANDIDATE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```

create separate method checkTimeLivingInCountry